### PR TITLE
Refactor Seekbar code

### PR DESCRIPTION
**Pull request by NikhilNarayana**
* Move SlippiPlayback into Core
* Pull out most of the playback specific variables and functions from the EXI device into SlippiPlaybackStatus
* Rename slippiTimer -> SlippiTimer
* Fix a stack corruption in SlippiTimer (sprintf puts a null terminator at the end of the string)
* Make the seekbar hideable (`-hs` flag or via the View menu)
* Update watch settings if the user seeks in a better manner (specifically to alleviate some issues that might occur when seeking during a queue of replays with specific startFrame/endFrame)
  * If the user seeks within the startFrame and endFrame then nothing changes
  * If the user seeks to before startFrame then startFrame gets updated to the targetFrameNum
  * If the user seeks past endFrame then endFrame gets updated to INT_MAX
* General code clean up

### CODE REVIEW

File: Source/Core/Core/Slippi/SlippiReplayComm.cpp
Comment (JLaferri) [inline Source/Core/Core/Slippi/SlippiReplayComm.cpp]: what does this do?

Comment (NikhilNarayana) [reply Source/Core/Core/Slippi/SlippiReplayComm.cpp]: fixes a warning I kept getting about std::isspace not being a member

**Comment by NikhilNarayana** [comment]
probably good to merge once we confirm it builds on linux/macOS

File: Source/Core/SlippiPlayback/SlippiPlayback.h
Comment (eigenform) [inline Source/Core/SlippiPlayback/SlippiPlayback.h]: forward-slashes please!

**Review by eigenform — DISMISSED** [review]
1.There are a bunch of `#include`s in this changeset that use backslashes - Linux doesn't like this
2. My compiler doesn't like that the `u8` type isn't visible from `SlippiPlayback.h`; perhaps need an `#include "../Common/CommonTypes.h`

Also, should probably merge #28 before merging this.

File: Source/Core/SlippiPlayback/SlippiPlayback.h
Comment (eigenform) [inline Source/Core/SlippiPlayback/SlippiPlayback.h:41]: Needed to `#include "../Common/CommonTypes.h"` here in order to get this to build

**Comment by NikhilNarayana** [comment]
meta confirmed that Linux builds and works properly.

Just macOS left now.

**Comment by NikhilNarayana** [comment]
builds on Mojave without too much issue. Everything seems to work fine

d4ae24a86e532285b3e5beaa08b94010732c2d6c is based on my environment, it shouldn't break other envs but it was required for the latest macOS Mojave update.

**Comment by NikhilNarayana** [comment]
Closing in favor of #76

**Pull request closed without merging**
